packages/@ember/-internals/glimmer/tests/integration/modifiers/on-test.js

```diff
@@ -0,0 +1,324 @@
+        let count = 0;
+
+        this.render('<button {{on "click" this.callback once=true}}>Click Me</button>', {
+          callback() {
+            count++;
+          },
+        });
+
+        assert.equal(count, 0, 'not called on initial render');
+
+        this.assertStableRerender();
+        assert.equal(count, 0, 'not called on a rerender');
+
+        runTask(() => this.$('button').click());
+        assert.equal(count, 1, 'has been called 1 time');
+
+        runTask(() => this.$('button').click());
+        assert.equal(count, 1, 'has been called 1 times');
+
+        if (isIE11) {
+          this.assertCounts({ adds: 1, removes: 1 });
+        } else {
+          this.assertCounts({ adds: 1, removes: 0 });
+        }
+      }
+
+      '@test changing from `once=false` to `once=true` ensures the callback can only be called once'(
+        assert
+      ) {
+        let count = 0;
+
+        this.render('<button {{on "click" this.callback once=once}}>Click Me</button>', {
+          callback() {
+            count++;
+          },
+
+          once: false,
+        });
+
+        runTask(() => this.$('button').click());
+        assert.equal(count, 1, 'has been called 1 time');
+
+        runTask(() => this.$('button').click());
+        assert.equal(count, 2, 'has been called 2 times');
+
+        runTask(() => this.context.set('once', true));
+        runTask(() => this.$('button').click());
+        assert.equal(count, 3, 'has been called 3 time');
+
+        runTask(() => this.$('button').click());
+        assert.equal(count, 3, 'is not called again');
+
+        if (isIE11) {
+          this.assertCounts({ adds: 2, removes: 2 });
+        } else {
+          this.assertCounts({ adds: 2, removes: 1 });
+        }
+      }
+
+      '@test setting passive named argument prevents calling preventDefault'() {
+        let matcher = /You marked this listener as 'passive', meaning that you must not call 'event.preventDefault\(\)'/;
+        this.render('<button {{on "click" this.callback passive=true}}>Click Me</button>', {
+          callback(event) {
+            expectAssertion(() => {
+              event.preventDefault();
+            }, matcher);
+          },
+        });
+
+        runTask(() => this.$('button').click());
+      }
+
+      '@test by default bubbling is used (capture: false)'(assert) {
+        this.render(
+          `
+            <div class="outer" {{on 'click' this.handleOuterClick}}>
+              <div class="inner" {{on 'click' this.handleInnerClick}}></div>
+            </div>
+          `,
+          {
+            handleOuterClick() {
+              assert.step('outer clicked');
+            },
+            handleInnerClick() {
+              assert.step('inner clicked');
+            },
+          }
+        );
+
+        runTask(() => this.$('.inner').click());
+
+        assert.verifySteps(['inner clicked', 'outer clicked'], 'uses capture: false by default');
```

Comment:
I did not know about the [`verifySteps`](https://api.qunitjs.com/assert/verifySteps) API. Nice! 👏